### PR TITLE
MM 7971 dockerhost -> localhost && docker -> docker-compose

### PR DESCRIPTION
**Pull request by scottleedavis**
#### Summary
The request 
* removes all occurrences of dockerhost and replaces them with localhost
* replaces contents of start/stop-docker in the makefile with docker-compose calls

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/10121#issuecomment-492558716

### CODE REVIEW

**Comment by scottleedavis** [comment]
added ports exposure in docker-compose.yml as it is needed for OSX.  https://docs.docker.com/docker-for-mac/networking/

**Comment by scottleedavis** [comment]
@hanzei This is no longer work in progress.  :)

**Comment by hanzei** [comment]
Thanks Scott, I requested two dev reviews.

**Comment by scottleedavis** [comment]
@lieut-data here are some thoughts in terms of upgrade path.  Please let me know what should change!  :)

1. update config.json to use localhost for DataSource.  
2. stop/remove all docker containers using the old approach on any `make start/stop/clean'-docker` or `make nuke` maybe by another Makefile task called `make clean-old-docker` that is called on the above normal docker tasks ?  Open to other approaches too.

**Comment by lieut-data** [comment]
1. I think it's fine to leave `dockerhost` in place, if only because developers will still have their `dockerhost` alias in `/etc/hosts` anyway, so it won't stop working.
2. I like the idea of a `make clean-old-docker` -- opt-in, no magic -- along with a PSA in the developers channel to notify the world. You can put this target in `build/legacy.mk` to avoid cluttering the main `Makefile`. The only concern is that some developers might have data in their existing databases, and it might be annoying to migrate. Is there a way to "rename" the existing docker containers into what docker compose might bring up for itself?

**Comment by scottleedavis** [comment]
I pushed the changes for legacy.mk.

I also looked into reusing containers from docker to docker-compose.  I had originally thought that by using the same container name (e.g. `container_name: "mattermost-mysql"`) would solve this, but docker-compose errors with trying to create the same name.
```
ERROR: for mysql  Cannot create container for service mysql: Conflict. The container name "/mattermost-mysql" is already in use by container "79f6efd6116cd988e5abc2e5d9fcf7705242a1170ad7ad39894ed94ae89b2f83". You have to remove (or rename) that container to be able to reuse that name.
```
This could likely be overcome likely by similar logic as exists in the makefile with `docker ps ` before running docker-compose.  

I agree that for some a complete loss of the containers would be a pain and would like to find a good solution to this.  @cpanato do you have thoughts on a reuse of containers between docker and docker-compose?

**Comment by scottleedavis** [comment]
@lieut-data if it is not possible to reuse the image, would a data migration script be appropriate?

**Comment by lieut-data** [comment]
I'm somewhat of the mindset that maaaybe we don't care enough to fix the migration, since it's a one-time pain? Perhaps something I can raise at the developer's meeting on your behalf later today to get the team's input.

**Comment by lieut-data** [comment]
@scottleedavis, see discussion at https://community-daily.mattermost.com/core/pl/5bmr8uxg4i8sdjkaatb9jjsexh, but I think we're good to skip any built-in migration and instead document the following:
* how to move from the "old way" to the "new way"
* how to optionally migrate your data between containers (e.g. could just be a bash pipe with a mysql export to a file, then a bash pipe to import when the new containers are started)

Are you up for submitting a PR to developers.mattermost.com to help with same?

**Comment by scottleedavis** [comment]
@lieut-data I haven't submitted a PR to developers.mattermost.com yet, but definitely up for it.

**Comment by lieut-data** [comment]
Excellent! Let me know if I can help, and we can get merge these changes together.

**Comment by hanzei** [comment]
@lieut-data @cpanato The other PR is ready to merge. Would you please re-review this?

**Comment by lieut-data** [comment]
LGTM -- @cpanato, once you've had a look, lets merge both PRs.

**Comment by scottleedavis** [comment]
@cpanato I know you are busy, however when you have time your thoughts/review is greatly appreciated.

**Comment by lieut-data** [comment]
Switching out @cpanato (busy) for @crspeller -- Christopher, I know you've already chimed in on some of this at the developer meeting, so I thought it might be good for you to take a final pass for any concerns.

File: Makefile [hunk 1]
```diff
@@ -116,188 +116,25 @@ start-docker: ## Starts the docker containers for local development.
...
-	fi
-
-	@if [ $(shell docker ps -a --no-trunc --quiet --filter name=^/mattermost-elasticsearch$$ | wc -l) -eq 0 ]; then \
-		echo starting mattermost-elasticsearch; \
-		docker run --name mattermost-elasticsearch -p 9200:9200 -e "http.host=0.0.0.0" -e "transport.host=127.0.0.1" -e "ES_JAVA_OPTS=-Xms250m -Xmx250m" -d mattermost/mattermost-elasticsearch-docker:6.5.1 > /dev/null; \
-	elif [ $(shell docker ps --no-trunc --quiet --filter name=^/mattermost-elasticsearch$$ | wc -l) -eq 0 ]; then \
-		echo restarting mattermost-elasticsearch; \
-		docker start mattermost-elasticsearch> /dev/null; \
-	fi
+	docker-compose --no-ansi -f ./build/docker-compose.yml run --rm start_dependencies
```
Comment (crspeller) [inline Makefile]: We have removed the already running checks. Does it work fine if run multiple times?

Comment (scottleedavis) [reply Makefile]: @crspeller `make start-docker` runs fine when run multiple times.  The containers are checked if they are already running, and if not starts them.  Was this your question?

Comment (crspeller) [reply Makefile]: Yup. As long as it can be run multiple times it should be fine.

**Comment by cpanato** [comment]
@crspeller PTAL

**Comment by lieut-data** [comment]
Thanks for your patience on this one, @scottleedavis! Looking forward to the simpler setup :)

**Comment by scottleedavis** [comment]
Jenkins is failing on the `Clean checkout` step.  Is this something I have control over fixing?  Haven't seen this one before.

**Comment by hanzei** [comment]
@scottleedavis I re-run the CI and it's green now.

@lieut-data Would you please merge this if you don't have any outstanding concerns?

**Pull request merged**